plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveBucketProperty.java

```diff
@@ -66,11 +67,18 @@ public static Optional<HiveBucketProperty> fromStorageDescriptor(Map<String, Str
         List<SortingColumn> sortedBy = ImmutableList.of();
         if (storageDescriptor.isSetSortCols()) {
             sortedBy = storageDescriptor.getSortCols().stream()
-                    .map(order -> SortingColumn.fromMetastoreApiOrder(order, tablePartitionName))
+                    .map(order -> SortingColumn.fromMetastoreApiOrder(order, tablePartitionName, dataColumns))
                     .collect(toImmutableList());
         }
         BucketingVersion bucketingVersion = HiveBucketing.getBucketingVersion(tableParameters);
-        return Optional.of(new HiveBucketProperty(storageDescriptor.getBucketCols(), bucketingVersion, storageDescriptor.getNumBuckets(), sortedBy));
+        List<String> bucketColumnNames = storageDescriptor.getBucketCols().stream()
+                // Ensure that the names used for the bucket columns are the same as the names used for the table columns
+                .map(bucketColumnName -> dataColumns.stream().filter(column -> column.getName().equalsIgnoreCase(bucketColumnName))
+                        .findFirst()
```

Comment:
there should be exactly one such

also, linear search over column list is OK provided that there are only few bucketing columns, which is a reasonable assumption (otherwise we would build a set of data column names). add a code comment.

however, i wonder whether we have to do this validation here at all
i think it should be sufficient to lowercase the column names.

```
storageDescriptor.getBucketCols().stream()
 .map(name -> name.toLowerCase(ENGLISH))
 .collect(toImmList())
```